lib/ACL/ACLManager.php

```diff
@@ -98,13 +99,22 @@ private function getRules(array $paths, bool $cache = true): array {
 	 * @return string[]
 	 */
 	private function getRelevantPaths(string $path): array {
-		$paths = [$path];
+		$paths = [];
+		$fromTrashbin = str_starts_with($path, '__groupfolders/trash/');
+		if ($fromTrashbin) {
+			$rootName = explode('/', $path, 5)[3];
+			$rootName = substr($rootName, 0, strrpos($rootName, '.d'));
```

Comment:
Too generic, will fail for `.doc`, or any folder with `.d` in it, no?

Reply:
```suggestion
			$rootName = preg_replace('/(.d[0-9]+)$/', '', $rootName);
```

Reply:
> Too generic, will fail for `.doc`, or any folder with `.d` in it, no?

No, strrpos takes the last occurrence.